### ISSUE TITLE
Indexing of Blacklight pages by Google or other search engines

### ISSUE THREAD

**Issue by flapka** [ycba-cia/blacklight-collections2#284]
As far as I can tell, item pages in our Blacklight catalog are not currently indexed by Google or other search engines. Could we do something to allow/encourage such crawls? 

When our catalog was based in Vufind, quite a few researchers came to our material via Google searches, and it'd be a shame to lose this point of entry.

**Comment by yulgit1** [comment]
There might be a few things to work out (where to deploy, what to disallow, how often to update) but here is a wiki page describing an implementation process:

https://github.com/projectblacklight/blacklight/wiki/Search-engine-harvesting

**Comment by yulgit1** [comment]
https://github.com/ycba-cia/blacklight-collections2/commit/30cf15a269bed45ddf1da1d60bb16e4aa12a6933
I added sitemaps leveraging solr to create an entry for each of the 119148 object in blacklight in 3 "sub" sitemaps as there is a 50000 max per sitemap. With it is the robots.txt referring to the master sitemap and disallowing things were don't want indexed like search results and facets per the above documentation.  

Will have to see if the item pages start showing up in google. FWIW before hand, I did a search on "ycba Prue" and got back the vufind which of course has been set up to redirect to blacklight for B1977.14.67.

```
SitemapGenerator::Sitemap.default_host = 'https://collections.britishart.yale.edu'
SitemapGenerator::Sitemap.create do

solr = RSolr.connect :url => "https://user:*****@ciaindex.britishart.yale.edu/solr/ycba_blacklight_prod"

response = solr.post 'select', :params => { # you may need to change the request handler
      'q'          => '',
      'fl'         => 'id', # we only need the ids
      'fq'         => '', # optional filter query
      'start'      => 0,        
      'rows'       => 150000,
      'sort'       => 'id asc'
    }

response['response']['docs'].each do |doc|
      add "/catalog/#{doc['id']}"
      puts doc['id']
end

end

SitemapGenerator::Sitemap.ping_search_engines
```

**Comment by edgartdata** [comment]
YCBA's accession numbers of the 5,000+ art collection objects that were contributed to the Google Art Project back in 2011 were indexed by Google, which allowed people to type B1977.14.67 directly in the Google search bar to get to its detailed record in the YCBA online catalog (and the redirect from Vufind to BlackLight works great).
These 5,000+ accession numbers are still indexed in such a way by Google today but not the latest 11,000+ additions to the Google Art & Culture platform. Should we pursue this further with our Google contacts?

![YCBA accession numbers indexed by Google](https://user-images.githubusercontent.com/4093644/102903754-c513c380-443e-11eb-8209-9e331fd1363f.png)

**Comment by yulgit1** [comment]
I've been monitoring this with googling the following query to just see blacklight and not vufind:
`site:collections.britishart.yale.edu -vufind`
About 2,210 results (12/14 mon night)
About 2,260 results (tue morning)
About 2,320 results (wed morning)
About 2,900 results (thu morning)
About 255,000 results (tue morning)

The bump up to 255,000 is great, which is what I was hoping for, although doesn't the catalog only have 120,000 items?  The first 10 pages are all blacklight catalog pages as they should be.

For B numbers I searched google for B1970.1 and got our blacklight page for Elihu as second hit.  I tried a more recent B2019.14.1 and didn't find anything.  Given that blacklight is 2020, I don't think what we gave them in 2011 would apply to blacklight B1970.1?  That is to say I'm unclear of the mechanism google uses to allow searches for B numbers.

**Comment by yulgit1** [comment]
Yes I would vote for collections.britishart.yale.edu/print to be disallowed.  Maybe even try to take steps to remove it. 

BTW the /catalog count grew a bit over the past 3 hours (now up 49,400).  Seems like a big increase in a short time.

**Comment by yulgit1** [comment]
I added Noindex to the /print pages, which means the next time crawled (whenever google gets to it) these pages should be removed.

I wanted to do the same for /vufind pages (we're redirectly already, but to avoid the redundancy), but as it is redirecting, this doesn't work.  Instead I disallowed these pages meaning crawling will cease for this pages, but not necessarily remove them, but that may occur over an indeterminate time.

addressed in:
https://github.com/ycba-cia/blacklight-collections2/commit/dd0e73edb1343a74ea843aa264183286752f5bfa

Also concerned about google indexing collections-test was thinking of adding a canonical header to point to production:

link rel="canonical" href="https://collections.britishart.yale.edu<%= request.path %>" 

However google has not yet found collections-test (which makes sense as these pages are probably not in any crawlable link, and there is no sitemap for these).  So decided against this, not wanting to create any problems.

https://www.gsqi.com/marketing-blog/dangerous-rel-canonical-problems/

**Comment by flapka** [comment]
As of today, the search "site:collections.britishart.yale.edu/catalog/ -vufind" yields:
* 25,900 hits in Google (the number is down presumably because we removed the /print pages?)
* 61,500 hits in Bing (if that even matters)

Do we want to consider this resolved, since full indexing is beyond our control?
